### PR TITLE
Qualcomm AI Engine Direct - LLAMA2 Infrastructure

### PR DESCRIPTION
**Pull request by chunit-quic**
1. OPs
- Add pow_tensor_scalar op
- Add rsqrt op
- Add sigmoid op
- Refine axis handling of cat op
- Refine parameters related functions
2. Passes
- Add AnnotateDecomposed for unbind and stak op
- Add DecomposeSilu for quantizer
- Add ReplaceInfBuffer for quantizer
- Change pass name ConvertAddmmmmWithLinear to ConvertToLinear
- Change pass name ConvertScaledDotProductAttention to DecomposeScaledDotProductAttention
- Support more args for sdpa op in DecomposeScaledDotProductAttention
- Support mm case for ConvertToLinear
- Move q_ops and dq_ops to pass/utils.py
3. Tests
- Add dummy llama2 test script
- Add single op test cases
4. Others
- Fix error of popping missing buffer
- Reorder the order of test models
- Reorder the order of op in qnn_constant

### CODE REVIEW

**Review by cccclai — APPROVED** [review]

File: exir/backend/backend_api.py [hunk 1]
```diff
@@ -250,7 +250,10 @@ def _partition_and_lower_one_graph_module(
                 # Delete the consumed buffers
                 buffer_name = toplevel_signature.inputs_to_buffers.pop(node.name)
                 toplevel_signature.buffers.remove(buffer_name)
-                owning_program.state_dict.pop(buffer_name)
+                if buffer_name in owning_program.state_dict:
```
Comment (cccclai) [inline exir/backend/backend_api.py:253]: Oh thanks for catching this case
